src/promptflow-tools/promptflow/tools/common.py

```diff
@@ -82,7 +82,7 @@ def validate_functions(functions):
 
 
 def parse_function_role_prompt(function_str):
-    pattern = r"\n*name:\n\s*(\S+)\s*\n*content:\n(.*)"
+    pattern = r"\n*#{0,2}\s*name:\n\s*(\S+)\s*\n*#{0,2}\s*content:\n(.*)"
```

Comment:
also comment here, to denote 2 for heading 2 format and 0 for backward compatibility.